### PR TITLE
use gcc cleanup attribute to refactor the commom module code and form…

### PR DESCRIPTION
**Pull request by JingWoo**
…at it
```bash
➜  libocispec git:(master) ✗ make check
make: Circular /root/workspace/third-party/libocispec/tests/data <- /root/workspace/third-party/libocispec/tests/data dependency dropped.
  GEN      public-submodule-commit
make  check-am
make[1]: Entering directory '/root/workspace/third-party/libocispec'
make  check-TESTS
make[2]: Entering directory '/root/workspace/third-party/libocispec'
make[3]: Entering directory '/root/workspace/third-party/libocispec'
PASS: tests/test-1
PASS: tests/test-2
PASS: tests/test-3
PASS: tests/test-4
PASS: tests/test-5
PASS: tests/test-6
PASS: tests/test-7
PASS: tests/test-8
PASS: tests/test-9
============================================================================
Testsuite summary for libocispec 0.2
============================================================================
# TOTAL: 9
# PASS:  9
# SKIP:  0
# XFAIL: 0
# FAIL:  0
# XPASS: 0
# ERROR: 0
============================================================================
make[3]: Leaving directory '/root/workspace/third-party/libocispec'
make[2]: Leaving directory '/root/workspace/third-party/libocispec'
make[1]: Leaving directory '/root/workspace/third-party/libocispec'
```
Signed-off-by: wujing <wujing50@huawei.com>

### CODE REVIEW

**Comment by giuseppe** [comment]
thanks, the change LGTM but please keep the original indentation.

For libocispec we use the GNU indentation: https://www.gnu.org/prep/standards/html_node/Formatting.html

**Comment by JingWoo** [comment]
OK, I have revised the code

File: src/common_c.py [hunk 1]
```diff
@@ -495,8 +495,8 @@ def generate_json_common_c(out):
 		  (long long int) map->keys[i]);
       if (nret < 0 || (size_t) nret >= sizeof (numstr))
 	{
-	  if (!*err && asprintf (err, "Error to print string") < 0)
-	    *(err) = strdup ("error allocating memory");
+	  if (!*err)
+            *err = strdup("Error to print string");
```
Comment (giuseppe) [inline src/common_c.py:499]: please add a space between strdup and '('

File: src/common_c.py [hunk 2]
```diff
@@ -729,70 +738,71 @@ def generate_json_common_c(out):
...
           if (invalid)
             {
-              free_json_map_int_bool (ret);
-              if (*err == NULL)
-                asprintf (err, "Invalid key '%s' with type 'int': %s",
-                          srckey, strerror (-invalid));
+              if (*err == NULL && asprintf (err, "Invalid key '%s' with type 'int': %s",
+                    srckey, strerror (-invalid)) < 0)
+                {
+                  *err = strdup("error allocating memory");
```
Comment (giuseppe) [inline src/common_c.py:772]: same here

File: src/common_c.py [hunk 3]
```diff
@@ -729,70 +738,71 @@ def generate_json_common_c(out):
...
           else
             {
-              free_json_map_int_bool (ret);
-              if (*err == NULL)
-                asprintf (err, "Invalid value with type 'bool' for key '%s'",
-                          srckey);
+              if (*err == NULL && asprintf (err,
+                    "Invalid value with type 'bool' for key '%s'", srckey) < 0)
+                {
+                  *err = strdup("error allocating memory");
```
Comment (giuseppe) [inline src/common_c.py:789]: and here

File: src/common_c.py [hunk 4]
```diff
@@ -853,7 +864,7 @@ def generate_json_common_c(out):
       if (nret < 0 || (size_t) nret >= sizeof (numstr))
 	{
 	  if (!*err)
-            asprintf (err, "Error to print string");
+            *err = strdup("Error to print string");
```
Comment (giuseppe) [inline src/common_c.py:867]: another case

File: src/common_c.py [hunk 5]
```diff
@@ -959,16 +971,19 @@ def generate_json_common_c(out):
...
           if (!YAJL_IS_STRING (srcval))
             {
-              free_json_map_int_string (ret);
-              if (*err == NULL)
-                asprintf (err, "Invalid value with type 'string' for key '%s'",
-                          srckey);
+              if (*err == NULL && asprintf (err,
+                   "Invalid value with type 'string' for key '%s'", srckey) < 0)
+                {
+                  *err = strdup("error allocating memory");
```
Comment (giuseppe) [inline src/common_c.py:977]: another here

File: src/common_c.py (hunk 1 above)
Comment (JingWoo) [reply src/common_c.py]: all done

**Comment by giuseppe** [comment]
thanks!

**Pull request merged**